### PR TITLE
feat(o-shop): adds login functionality, simulated backend

### PR DESCRIPTION
**Pull request by mcraepv**
feat(o-shop): creates login, register forms

feat(o-shop): adds login and register forms, updates for programmatic form generation

feat(o-shop): adds login functionality, simulated backend

removes comment

feat(o-shop): adds login state and simulates backend functionality

### CODE REVIEW

File: frontend/src/components/Form/Form.js [hunk 1]
```diff
@@ -23,15 +23,17 @@ const Form = (props) => {
 
   const onSubmit = (event) => {
     event.preventDefault();
-    //checking for empty fields
+
     setSubmitted(true);
+
+    //checking for empty fields
```
Comment (BarretteBanner) [inline frontend/src/components/Form/Form.js]: please remove comment

Comment (mcraepv) [reply frontend/src/components/Form/Form.js]: removed

File: frontend/src/store/store.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { createStore, combineReducers, applyMiddleware } from 'redux';
+import loginReducer from './login/reducers';
+// import registerReducer from './register/reducer';
```
Comment (BarretteBanner) [inline frontend/src/store/store.js]: comments here and a few other places like line 13. I'm okay with documentation just not commented out code

File: frontend/src/store/store.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { createStore, combineReducers, applyMiddleware } from 'redux';
```
Comment (BarretteBanner) [inline frontend/src/store/store.js:1]: also we like to use the convention of 3rd party imports, blank line, imports from other libraries within your code base (you don't have any rn), blank line, then local imports (anything that starts with ./ or ../)
Please update all your js files to follow this convention

File: frontend/src/services/userService.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+      return user;
+    });
+};
+
+const logout = () => {
+  localStorage.removeItem('user');
+};
+
+const register = (user) => {
```
Comment (BarretteBanner) [inline frontend/src/services/userService.js]: this function doesn't do anything remove it

File: frontend/src/utils/FakeBackend.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+              .catch((error) => reject(error));
+        }
+      };
+      // wrap in timeout to simulate server api call
+      setTimeout(handleRoute, 500);
+      // route functions
+
+      const authenticate = () => {
+        const { username, password } = body;
+        const user = users.find((x) => {
```
Comment (BarretteBanner) [inline frontend/src/utils/FakeBackend.js]: no abbreviations, fix here and across your code base

**Review by BarretteBanner — APPROVED** [review]

**Pull request merged**